SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -1128,31 +1128,52 @@ Plotter {
 
 + Function {
 	plot { |duration = 0.01, target, bounds, minval, maxval, separately = false|
+		var server, plotter, action;
+		var name = this.asCompileString;
+
+		if(name.size > 50 or: { name.includes(Char.nl) }) { name = "function plot" };
 
-		var name = this.asCompileString, plotter, action;
-		if(name.size > 50 or: { name.includes(Char.nl) }) { name = "Function plot" };
 		plotter = Plotter(name, bounds);
+		// init data in case function data is delayed (e.g. server booting)
 		plotter.value = [0.0];
+
 		target = target.asTarget;
+		server = target.server;
 		action = { |array, buf|
 			var numChan = buf.numChannels;
-			{
+			var numFrames = buf.numFrames;
+			var frameDur;
+
+			defer {
 				plotter.setValue(
-					array.unlace(numChan).collect(_.drop(-1)),
+					array.unlace(numChan),
 					findSpecs: true,
 					refresh: false,
 					separately: separately,
 					minval: minval,
 					maxval: maxval
 				);
+
 				plotter.domainSpecs = ControlSpec(0, duration, units: "Time (s)");
-			}.defer
+
+				// If individual values might be separated by at least 2.5 px
+				// (based on a plot at full screen width), set the x values (domain)
+				// explicitly for accurate time alignment with grid lines.
+				if(numFrames < (Window.screenBounds.width / 2.5)) {
+					frameDur = (this.value.rate == \control).if({
```

Comment:
`frameDur = if(this.value.rate == \control) {`  would keep one style
Also no need for semicolons for the return of the functions (e.g. for `1 / server.sampleRate;`)

Reply:
(but this is just a suggestion)

Reply:
Thanks for suggesting this. I thought at the time that setting a variable with receiver syntax read more naturally, but I agree consistency is better.

After reviewing the [discussion](https://github.com/supercollider/supercollider/issues/2914) that led to return statement [guidelines](https://github.com/supercollider/supercollider/wiki/Code-style-guidelines#return-statements), I removed `;` from both return statements in the `if`, and added one after `defer` (that statement doesn't imply return and could be followed by more code).